esiclient/v1/node_network.py

```diff
@@ -54,23 +54,46 @@ def take_action(self, parsed_args):
         if parsed_args.network:
             filter_network = neutron_client.find_network(parsed_args.network)
 
+        nodes = ironic_client.node.list()
+        neutron_ports = neutron_client.ports()
+        networks = neutron_client.networks()
```

Comment:
I think these can be reduced; if parsed_args.node is defined, then we only need to fetch one node; similarly if parsed_args.network is defined. Neutron ports can also be filtered by networks if parsed_args.network is defined.

Reply:
Fixed.